jsapp/js/components/special/koboAccessibleSelect.spec.tsx

```diff
@@ -0,0 +1,136 @@
+    expect(listOptions).toHaveLength(3);
+  });
+
+  it('should be selectable by mouse click', async () => {
+      const trigger = screen.getByRole('combobox');
+      const list = screen.getByRole('listbox');
+      const listOptions = screen.getAllByRole('option');
+
+      // Clicks the trigger
+      await user.click(trigger);
+      expect(list.dataset.expanded).toBe('true');
+
+      // Select first option
+      await user.click(listOptions[0]);
+
+      // Onchange should be called with the correct value
+      expect(onChangeMock).lastCalledWith(options[0].value);
+
+      expect(list.dataset.expanded).toBe('false');
+  });
+
+  it('should be selectable by keyboard arrows', async () => {
+      const trigger = screen.getByRole('combobox');
+      const triggerLabel = trigger.querySelector('label');
+
+      // No item selected
+      expect(triggerLabel).toHaveTextContent('Select…');
+
+      // Increase option on arrow down
+      fireEvent.keyDown(trigger, {key: 'ArrowDown'});
+      expect(onChangeMock).lastCalledWith(options[0].value);
+      expect(triggerLabel).toHaveTextContent(options[0].label);
+
+      // Increase option on arrow right
+      fireEvent.keyDown(trigger, {key: 'ArrowRight'});
+      expect(onChangeMock).lastCalledWith(options[1].value);
+      expect(triggerLabel).toHaveTextContent(options[1].label);
+      fireEvent.keyDown(trigger, {key: 'ArrowRight'});
+      expect(onChangeMock).lastCalledWith(options[2].value);
+      expect(triggerLabel).toHaveTextContent(options[2].label);
+
+      // Don't go past the last one
+      onChangeMock.mockReset();
+      fireEvent.keyDown(trigger, {key: 'ArrowDown'});
+      expect(onChangeMock).not.toHaveBeenCalled();
+      expect(triggerLabel).toHaveTextContent(options[2].label);
+
+      // Decrease option on arrow up
+      fireEvent.keyDown(trigger, {key: 'ArrowUp'});
+      expect(onChangeMock).lastCalledWith(options[1].value);
+      expect(triggerLabel).toHaveTextContent(options[1].label);
+
+      // Decrease option on arrow left
+      fireEvent.keyDown(trigger, {key: 'ArrowLeft'});
+      expect(onChangeMock).lastCalledWith(options[0].value);
+      expect(triggerLabel).toHaveTextContent(options[0].label);
+
+      // Don't go past the first one
+      onChangeMock.mockReset();
+      fireEvent.keyDown(trigger, {key: 'ArrowUp'});
+      expect(onChangeMock).not.toHaveBeenCalled();
+      expect(triggerLabel).toHaveTextContent(options[0].label);
+  });
+
+  it('should be selectable by typing', async () => {
+      const trigger = screen.getByRole('combobox');
+      const triggerLabel = trigger.querySelector('label');
+
+      // No item selected
+      expect(triggerLabel).toHaveTextContent('Select…');
+
+      // Type 'b' to select Banana
+      fireEvent.keyDown(trigger, {key: 'b'});
+      expect(onChangeMock).lastCalledWith(options[1].value);
+      expect(triggerLabel).toHaveTextContent(options[1].label);
+  });
+});
```

Comment:
Awesome! Great demo.

Seeing this makes me think of a bunch of other test cases. :) 

Looking forward to using jest tests to get more safety for stuff we'd test manually, including complicated components like the data table, and query logic if we need it (with [msw](https://github.com/testing-library/react-testing-library#:~:text=We%20recommend%20using%20Mock%20Service%20Worker%20library%20to%20declaratively%20mock%20API%20communication%20in%20your%20tests%20instead%20of%20stubbing%20window.fetch%2C%20or%20relying%20on%20third%2Dparty%20adapters.))